src/internalEnforcer.ts

```diff
@@ -92,7 +108,22 @@ export class InternalEnforcer extends CoreEnforcer {
   /**
    * updatePolicyInternal updates a rule from the current policy.
    */
-  public async updatePolicyInternal(sec: string, ptype: string, oldRule: string[], newRule: string[]): Promise<boolean> {
+  protected async updatePolicyInternal(sec: string, ptype: string, oldRule: string[], newRule: string[]): Promise<boolean> {
+    const res = this.updatePolicyInternalWithoutNotify(sec, ptype, oldRule, newRule);
+
+    if (this.watcher && this.autoNotifyWatcher) {
```

Comment:
Misses watcherEx?

Reply:
There was no `updateForUpdatePolicy` in watcherEx interface so I did not call it also I did not made a method like that in the watcherEx we are working on, should we update the interface because I guess `updateForUpdatePolicy` exists in go-casbin.

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

Reply:
Can we call the `UpdateForAddPolicies()` and `UpdateForRemovePolicies()` methods?

Reply:
I can merge this, but we need to make an issue to talk about this on the golang-casbin repository

Reply:
@nodece are you saying we call `UpdateForAddPolicies()` and `UpdateForRemovePolicies()` in updateInternalPolicy or make a watcher method for it?

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Tags:** Inefficient

Reply:
@nodece I have added changes for the above idea, the tests are failing rn but we will fix it once you aprove the idea.

Reply:
Or `UpdateForUpdatePolicy`? Make sense?